### PR TITLE
Monitor timeline bug fix

### PR DESCRIPTION
**Pull request by zhostasa**
Issue 434 - Concurrent modification exception and timetine bug fix 
 * POM dependency fix (for sake of Eclipse)
 * MasterMonitors added as MasterListener and stopped on configuration end
 * MasterListener.Adapter removed in favor of Java 8 default interface implementation
 * Comment added

 Overenthusiastic refactoring
 * United listeners under StateListener interface
 * Moved Listener management under StageBase class
 * Moved shared functionality from Monitors to AbstractMonitors class
 * Replaced interface Adapters with default interface implementation
 * Added some javadocs
 * Refactoring on MonitorStartStage

The first commit gets rid of the #434 and #442, which are related
The second commit is minor refactoring attempt gone somewhat more enthusiastic than originally intended. Especially in case of AbstractMonitors class I am not absolutely sure I did a right thing and hope for some input, bless my heart...

### CODE REVIEW

File: extensions/jpa/pom.xml [hunk 1]
```diff
@@ -18,6 +19,11 @@
          <artifactId>radargun-core</artifactId>
          <version>${project.version}</version>
       </dependency>
+      <dependency>
```
Comment (rvansa) [inline extensions/jpa/pom.xml]: again this eclipse thing?

**Comment by rvansa** [comment]
Maybe purpose is not the right word... But I find merging them as overgeneralization. And btw., keep commit titles a bit more 'professional', please :)

File: extensions/jpa/pom.xml (hunk 1 above)
Comment (zhostasa) [inline extensions/jpa/pom.xml]: This is the last time I do this. Until next time, of course.

**Comment by zhostasa** [comment]
I agree that they are hardly doing the same thing, but their usage is so similar that putting them under one interface and pushing the functionality to parent class, while not exactly pretty, seemed like a slightly better approach. That said, I am hardly authority on the project as a whole, so if you decide these changes are not beneficial I will remove them.

I will do the best ape can do to remain professional

File: extensions/jpa/pom.xml (hunk 1 above)
Comment (alanfx) [inline extensions/jpa/pom.xml]: Don't pick on us Eclipse users, @rvansa!

Comment (rvansa) [inline extensions/jpa/pom.xml]: I would never mock elderly people!

File: core/src/main/java/org/radargun/state/SlaveState.java [hunk 2]
```diff
@@ -68,7 +65,7 @@ public int getGroupSize() {
    public void reset() {
       super.reset();
       traits = null;
-      serviceListeners.clear();
+      clearlisteners();
```
Comment (rvansa) [inline core/src/main/java/org/radargun/state/SlaveState.java]: this should go to super.reset(), shouldn't it?

File: extensions/jpa/pom.xml (hunk 1 above)
Comment (alanfx) [inline extensions/jpa/pom.xml]: Some dinosaurs are deadly!

File: core/src/main/java/org/radargun/state/SlaveState.java (hunk 2 above)
Comment (zhostasa) [inline core/src/main/java/org/radargun/state/SlaveState.java]: Right, missed that one

**Comment by zhostasa** [comment]
Fixed a minor issue

**Pull request merged**